helpers/setURLQueryParam.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+const URL = require('url').URL;
+const utils = require('handlebars-utils');
+const common = require('./lib/common.js');
+const SafeString = require('handlebars').SafeString;
+
+const factory = () => {
+    return function(string, key, value) {
+        if (string instanceof SafeString) {
+            string = string.toString();
+        }
+        if (!utils.isString(string) || !common.isValidURL(string)
```

Comment:
I think we should break this check into multiple. If `string` is not a string or url we should throw a type error `throw new TypeError("Not a valid url")`